### ISSUE TITLE
Speed up interface to SIAF information (mostly for NIRCam)

### ISSUE THREAD

**Issue by Johannes-Sahlmann** [spacetelescope/mirage#202]
When generating the observation yamls, the SIAF xml is read again and again for every NIRCam detector (`self.subarray_bounds = siaf_interface.get_siaf_information(...)`). This takes considerable time in particular programs with many exposures.
This could be made faster by pre-loading the SIAF (e.g. of all instruments present in an APT program) and accessing these on an aperture basis.

**Comment by bhilbert4** [comment]
I can have a look at this, although I'll wait until after we merge #172 to avoid creating conflicts.

**Comment by bhilbert4** [comment]
I've started working on this. @Johannes-Sahlmann, in line 717 of apt_inputs.py, you have:
```
    if siaf_instrument == 'NIRSPEC':
                siaf_instrument = 'NIRSpec'
```

which is then used in a call to pysiaf. But there's no corresponding check for NIRCAM->NIRCam. Should that be there? It seems like it should, but I didn't get any errors about bad instrument names when testing for #172. Assuming the check should be there, I can add it as part of this update.

**Comment by Johannes-Sahlmann** [comment]
I guess there should be. Maybe for NIRCam this is caught somewhere upstream and that is why I did not get an error during my tests. 
In any case, this should soon become unnecessary, as I just put in https://github.com/spacetelescope/pysiaf/pull/70

**Comment by Johannes-Sahlmann** [comment]
pysiaf 0.2.5 (just released) is insensitive to instrument name capitalization (during initialization).
If you update the environment.yaml files, you could get rid of the 
```if siaf_instrument == 'NIRSPEC':
                siaf_instrument = 'NIRSpec'```
code.

**Comment by bhilbert4** [comment]
Solved by #216. Lines manipulating instrument names for input into pysiaf still remain. I'll create a new issue to update the environment files, at which point I think we can simplify the instrument name inputs to pyisaf.
